### PR TITLE
Init/v0.0.1

### PR DESCRIPTION
**Pull request by gagantrivedi**
This implements two endpoints:
1. To fetch feature state for a given feature and environment
2. To update feature state for a given environment

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+		Get(url)
+
+	if err != nil {
+		return nil, err
+	}
+	if !resp.IsSuccess() {
+		return nil, fmt.Errorf("Error getting feature state: %s", resp.Status())
+
+	}
+	featureState := result.Results[0]
```
Comment (matthewelwell) [inline client.go:52]: What about if there aren't any results? Won't this throw some kind of unhandled index error?

File: models.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+	if err != nil {
+		return err
+	}
+	fs.ID = obj.ID
+	fs.Enabled = obj.Enabled
+	fs.Feature = obj.Feature
+	fs.Environment = obj.Environment
+
+	// If the feature state value is a struct then unmarshal, set
```
Comment (matthewelwell) [inline models.go:42]: Can we explain the scenario here where FSV is / isn't a struct? I'm a bit lost on this.

File: client.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+		"Authorization": "Api-Key " + c.master_api_key,
+	})
+	return c
+
+}
+
+// Get the feature state associated with the environment for a given feature
+func (c *Client) GetEnvironmentFeatureState(environmentAPIKey string, featureName string) (*FeatureState, error) {
+	url := fmt.Sprintf("%s/environments/%s/featurestates/", c.baseURL, environmentAPIKey)
+	fmt.Println("making request with", url)
```
[the remark this answers is not in the document]
Comment (gagantrivedi) [reply client.go]: No, I removed it

File: client.go (hunk 1 above)
Comment (gagantrivedi) [reply client.go:57]: yep, my bad

File: models.go (hunk 2 above)
Comment (gagantrivedi) [reply models.go]: Yeah, I added the comment

File: client.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+// Update the feature state
+func (c *Client) UpdateFeatureState(featureState *FeatureState) (*FeatureState, error) {
+	url := fmt.Sprintf("%s/features/featurestates/%d/", c.baseURL, featureState.ID)
+	updatedFeatureState := FeatureState{}
+	resp, err := c.client.R().SetBody(featureState).SetResult(&updatedFeatureState).Put(url)
+	if err != nil {
+		return nil, err
+	}
+	if !resp.IsSuccess() {
```
[the remark this answers is not in the document]
Comment (gagantrivedi) [reply client.go:64]: Yeah, that's not idiomatic

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**